server/util/dsingleflight/dsingleflight.go

```diff
@@ -0,0 +1,195 @@
+// Package dsingleflight is a distributed version of the singleflight package.
+// Work is coordinated across servers using Redis.
+package dsingleflight
+
```

Comment:
nit: if you wanted to get really fancy you could add a comment with an example of how to use this

Reply:
done